dev/com.ibm.ws.concurrent.rx/src/com/ibm/websphere/concurrent/rx/ManagedCompletableFuture.java

```diff
@@ -246,12 +256,38 @@ private ManagedCompletableFuture(CompletableFuture<T> completableFuture, Executo
         return stage;
     }
 
+    /**
+     * Replaces CompletableFuture.delayedExecutor(delay, unit) with an implementation that uses the Liberty
+     * ScheduledExecutor for notification of the delay and captures thread context from the invoker of the execute method
+     * and propagates it to the running action. Except if the action is already contextualized, then the context of the
+     * action overrides.
+     *
+     * @param delay amount of time to delay
+     * @param unit time unit of the delay value
+     * @param executor executor upon which to run the task after the delay elapses
+     * @return new executor instance
+     */
+    @Trivial
     public static Executor delayedExecutor(long delay, TimeUnit unit) {
-        throw new UnsupportedOperationException(); // TODO implement when rebasing on Java 9
+        return delayedExecutor(delay, unit, AccessController.doPrivileged(getDefaultManagedExecutorAction));
```

Comment:
Should there be a check for Java8 in this method?

Reply:
No, it is unnecessary because the method to which it delegates does the Java 8 check.